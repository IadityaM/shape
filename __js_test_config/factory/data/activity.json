{"data":{"id":"1365201","type":"activities","attributes":{"action":null,"created_at":"2019-12-16T23:14:35.973Z","content":null,"actor_id":22548,"target_type":"Collection","target_id":762911,"source_name":""},"relationships":{"actor":{"meta":{"included":false}},"subject_users":{"meta":{"included":false}},"subject_groups":{"meta":{"included":false}},"target":{"meta":{"included":false}}}}}