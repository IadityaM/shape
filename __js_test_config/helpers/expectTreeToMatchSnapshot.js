import toJson from 'enzyme-to-json'

const expectTreeToMatchSnapshot = tree => {
  expect(toJson(tree)).toMatchSnapshot()
}

export default expectTreeToMatchSnapshot
