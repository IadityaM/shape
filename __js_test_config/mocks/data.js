export const fakeTextItemAttrs = {
  id: 1,
  type: 'Item::TextItem',
  name: 'My Cool Item',
  text_data: 'This is the content for the item and it contains multiple sentences. Like this one.',
  breadcrumb: [['collections', 1, 'Some collection'], ['items', 1, 'my item']],
  parentPath: '/',
}
export const fakeTextItem = {
  ...fakeTextItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeTextItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
  toJS: jest.fn().mockReturnValue(fakeTextItemAttrs),
}
export const fakeVideoItemAttrs = {
  id: 2,
  type: 'Item::VideoItem',
  name: 'Crazy Honey Badger',
  url: 'https://www.youtube.com/watch?v=4r7wHMg5Yjg'
}
export const fakeVideoItem = {
  ...fakeVideoItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeTextItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
}
export const fakeImageItemAttrs = {
  id: 3,
  type: 'Item::ImageItem',
  name: 'Earth from Space',
  filestack_file: {
    id: 1,
    url: 'https://www.nasa.gov/sites/default/files/styles/full_width_feature/public/thumbnails/image/iss052e023801_0.jpg',
  }
}
export const fakeImageItem = {
  ...fakeImageItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeTextItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
}
export const fakeItemCard = {
  id: 10,
  order: 0,
  height: 1,
  width: 1,
  record: fakeTextItem,
  item: fakeTextItem,
}
export const fakeCollectionCard = {
  id: 11,
  order: 1,
  height: 1,
  width: 1,
  record: fakeCollection,
  item: fakeCollection,
  reference: false,
}
export const fakeCollection = {
  id: 1,
  name: 'My Workspace X',
  type: 'Collection',
  breadcrumb: [],
  collection_cards: [
    fakeItemCard, fakeItemCard, fakeItemCard
  ]
}
export const fakePosition = {
  xPos: 0,
  yPos: 0,
  height: 1,
  width: 1,
}
