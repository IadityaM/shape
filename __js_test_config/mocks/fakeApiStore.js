const fakeApiStore = ({ findResult = '', findAllResult = [], requestResult = '' } = {}) => {
  return {
    add: jest.fn().mockName('add'),
    removeAll: jest.fn().mockName('removeAll'),
    find: jest.fn()
      .mockReturnValue(findResult),
    findAll: jest.fn()
      .mockReturnValue(findAllResult),
    request: jest.fn()
      .mockReturnValue(Promise.resolve(requestResult))
      .mockName('request'),
    fetch: jest.fn()
      .mockReturnValue(Promise.resolve())
      .mockName('fetch'),
    sync: jest.fn().mockName('sync'),
    setCurrentUserId: jest.fn(),
    currentUserId: 1,
    currentUser: {
      id: 1,
      current_user_collection_id: 99,
      current_organization: {
        name: 'test org'
      },
      name: 'Johnny Appleseed',
      pic_url_square: 'https://s3.amazonaws.com/pic.png',
      groups: [],
      terms_accepted: true,
    },
    collections: [],
    items: [],
  }
}

export default fakeApiStore
