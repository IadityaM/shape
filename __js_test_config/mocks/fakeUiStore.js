import { observable } from "mobx"

const fakeUiStore = {
  gridSettings: {
    cols: 4,
    gutter: 20,
    gridW: 312,
    gridH: 250,
  },
  defaultGridSettings: {
    cols: 4,
    gutter: 20,
    gridW: 312,
    gridH: 250,
  },
  blankContentToolState: {
    order: null,
    width: null,
    height: null,
    replacingId: null,
    placeholderCard: null,
  },
  dialogConfig: {
    open: null,
    prompt: null,
    onConfirm: null,
    onCancel: null,
    iconName: null,
    confirmText: 'OK',
    cancelText: 'Cancel',
    onClose: jest.fn(),
  },
  activityLogOpen: false,
  activityLogPosition: {
    x: 0,
    y: 0,
    h: 1,
    w: 1,
  },
  activityLogMoving: false,
  cardMenuOpen: {
    id: false,
    x: 0,
    y: 0,
    offsetX: 0,
    offsetY: 0,
  },
  selectedArea: { minX: null, maxX: null, minY: null, maxY: null },
  openBlankContentTool: jest.fn(),
  closeBlankContentTool: jest.fn(),
  setBctPlaceholderCard: jest.fn(),
  closeCardMenu: jest.fn(),
  openCardMenu: jest.fn(),
  closeMoveMenu: jest.fn(),
  resetSelectionAndBCT: jest.fn(),
  resetCardPositions: jest.fn(),
  rolesMenuOpen: false,
  isLoading: false,
  dismissedMoveHelper: false,
  selectedAreaEnabled: false,
  setSelectedArea: jest.fn(),
  selectedCardIds: [],
  selectCardId: jest.fn(),
  deselectCards: jest.fn(),
  setViewingRecord: jest.fn(),
  viewingCollection: null,
  isViewingHomepage: false,
  viewingItem: null,
  movingFromCollectionId: null,
  movingCardIds: [],
  openMoveMenu: jest.fn(),
  update: jest.fn(),
  alert: jest.fn(),
  alertOk: jest.fn(),
  defaultAlertError: jest.fn(),
  confirm: jest.fn(),
  closeDialog: jest.fn(),
  cardAction: 'move',
  blurContent: false,
  organizationMenuPage: 'organizationMenuPage',
  organizationMenuGroupId: null,
  organizationMenuOpen: false,
  expandedThreadKey: null,
  zoomLevel: 1,
  expandThread: jest.fn(),
  openGroup: jest.fn(),
  openOptionalMenus: jest.fn(),
  trackEvent: jest.fn(),
  trackedRecords: {},
  editingName: observable([]),
  activityLogPage: 'comments',
  pageMenuOpen: false,
  searchText: '',
  collectionCardSortOrder: '',
  addNewCard: jest.fn(),
  removeNewCard: jest.fn(),
  isNewCard: jest.fn(),
  autocompleteMenuClosed: jest.fn(),
  captureKeyboardGridClick: jest.fn(),
  popupAlert: jest.fn(),
  stopDragging: jest.fn(),
  multiMoveCardIds: [],
  setSnoozeChecked: jest.fn(),
  scrollToTop: jest.fn(),
  scrollToBottom: jest.fn(),
  scrollToBottomOfComments: jest.fn(),
  scrollToBottomOfModal: jest.fn(),
  popupSnackbar: jest.fn(),
  showPermissionsAlert: jest.fn(),
  gridHeightFor: jest.fn().mockReturnValue(250),
  performActionAfterRoute: jest.fn(),
  linkedBreadcrumbTrailForRecord: jest
    .fn()
    .mockImplementation(x => x.breadcrumb),
  addEmptySpaceClickHandler: jest.fn(),
  removeEmptySpaceClickHandler: jest.fn(),
  setEditingCardCover: jest.fn(),
  adminAudienceMenuOpen: false,
  actionMenuOpenForCard: jest.fn(true).mockImplementation(x => false),
  textMenuOpenForCard: jest.fn(true).mockImplementation(x => false),
  replyingToCommentId: false,
  setReplyingToComment: jest.fn(),
  setCommentingOnRecord: jest.fn(),
  setActivityLogPage: jest.fn(),
  setBodyBackgroundImage: jest.fn(),
  setBodyFontColor: jest.fn(),
  selectedTextRangeForCard: jest.fn(),
  reselectCardIds: jest.fn(),
  reselectOnlyEditableRecords: jest.fn(),
  reselectOnlyMovableCards: jest.fn(),
  setMovingCards: jest.fn(),
  setVisibleCols: jest.fn(),
  setVisibleRows: jest.fn(),
  setPlaceholderSpot: jest.fn(),
  setDroppingFilesCount: jest.fn(),
  drag: jest.fn(),
  startDragging: jest.fn(),
  adjustZoomLevel: jest.fn(),
  zoomIn: jest.fn(),
  zoomOut: jest.fn(),
  determineZoomLevels: jest.fn(),
  maxCols: jest.fn(),
  maxGridWidth: jest.fn().mockReturnValue(1384),
  pageMargins: jest.fn().mockReturnValue({ left: 100, top: 50 }),
  relativeZoomLevel: 2,
  zoomLevels: [],
  isSelected: jest.fn(),
  toggleSelectedCardId: jest.fn(),
  openContextMenu: jest.fn(),
  clearTextEditingItem: jest.fn(),
  positionForCoordinates: jest.fn().mockReturnValue({ x:0, y:0, xPos: 0, yPos: 0, width: 1, height: 1 }),
  dragGridSpot: new Map(),
  droppingFilesCount: 0,
}

export default fakeUiStore
