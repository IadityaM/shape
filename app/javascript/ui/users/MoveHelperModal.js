import PropTypes from 'prop-types'
import { observable, action } from 'mobx'
import { inject, observer, PropTypes as MobxPropTypes } from 'mobx-react'
import styled from 'styled-components'
import Dialog from '@material-ui/core/Dialog'
import DialogContent from '@material-ui/core/DialogContent'
import FormControl from '@material-ui/core/FormControl'
import FormControlLabel from '@material-ui/core/FormControlLabel'

import { FormButton, Checkbox } from '~/ui/global/styled/forms'
import v from '~/utils/variables'

const SpecialDisplayHeading = styled.p`
  text-align: center;
  margin: 0;
  margin-bottom: 30px;
  line-height: 1.625rem;
  font-family: ${v.fonts.sans};
  font-size: 1.25rem;
  font-weight: ${v.weights.book};
  color: ${v.colors.black};
`

const StyledDialog = styled(Dialog)`
  .modal__paper {
    text-align: center;
    padding: 20px;
    padding-top: 35px;
    max-width: 445px;
    min-width: 540px;
    width: 100%;
  }
`
StyledDialog.displayName = 'StyledDialog'

@inject('uiStore')
@observer
class MoveHelperModal extends React.Component {
  @observable
  dontShowChecked = false
  @observable
  isLoading = false
  @observable
  submitted = false

  @action
  handleDontShowCheck = event => {
    this.dontShowChecked = event.target.checked
  }

  @action
  handleSubmit = e => {
    e.preventDefault()
    const { currentUser, type, uiStore } = this.props
    this.submitted = true
    uiStore.update('dismissedMoveHelper', true)
    if (this.dontShowChecked) {
      currentUser.API_hideHelper(type)
    }
  }

  get helperText() {
    const { type, recordName } = this.props
    let text = ''
    if (type === 'move') {
      text = `
        Did you know when moving, duplicating, or linking items,
        you can navigate to another collection to place the items there?
      `
    } else if (type === 'template') {
      text = `
        Did you know? You can navigate to wherever you would like to place
        "${recordName}", and use the up or down arrows to place it at the top or
        bottom of the collection.
      `
    }
    return text
  }

  render() {
    return (
      <StyledDialog
        classes={{ paper: 'modal__paper' }}
        open={!this.submitted}
        BackdropProps={{
          invisible: true,
        }}
      >
        <DialogContent>
          <form onSubmit={this.handleSubmit}>
            <img
              src="https://s3-us-west-2.amazonaws.com/assets.shape.space/move_helper_diagram.png"
              alt="Diagram showing moving items between multiple collections"
              style={{ width: '410px', marginBottom: '40px' }}
            />
            <SpecialDisplayHeading>{this.helperText}</SpecialDisplayHeading>
            <FormControl component="fieldset" required>
              <FormControlLabel
                classes={{ label: 'form-control' }}
                control={
                  <Checkbox
                    checked={this.dontShowChecked}
                    onChange={this.handleDontShowCheck}
                    value="yes"
                  />
                }
                label="Thanks, please don't show me this message again."
              />
              <div style={{ height: '54px' }} />
            </FormControl>

            <div className="button--center">
              <FormButton disabled={this.isLoading}>Close</FormButton>
            </div>
          </form>
        </DialogContent>
      </StyledDialog>
    )
  }
}

MoveHelperModal.propTypes = {
  currentUser: MobxPropTypes.objectOrObservableObject.isRequired,
  recordName: PropTypes.string,
  type: PropTypes.string,
}

MoveHelperModal.wrappedComponent.propTypes = {
  uiStore: MobxPropTypes.objectOrObservableObject.isRequired,
}

MoveHelperModal.defaultProps = {
  type: 'move', // types are 'move' or 'template'
  recordName: null,
}

export default MoveHelperModal
