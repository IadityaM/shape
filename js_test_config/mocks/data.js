export const fakeItemAttrs = {
  id: 1,
  name: 'My Cool Item',
}
export const fakeItem = {
  ...fakeItemAttrs,
  rawAttributes: jest.fn().mockReturnValue(fakeItemAttrs),
  getRecordType: jest.fn().mockReturnValue('items'),
}
export const fakeCard = {
  id: 10,
  order: 0,
  height: 1,
  width: 1,
  record: fakeItem,
  item: fakeItem,
}
export const fakeCollection = {
  id: 1,
  name: 'My Workspace X',
  breadcrumb: [],
  collection_cards: [
    fakeCard, fakeCard, fakeCard
  ]
}
export const fakePosition = {
  xPos: 0,
  yPos: 0,
  height: 1,
  width: 1,
}
