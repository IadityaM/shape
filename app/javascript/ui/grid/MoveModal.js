import { inject, observer, PropTypes as MobxPropTypes } from 'mobx-react'
import { observable, runInAction } from 'mobx'
import styled from 'styled-components'

import CloseIcon from '~/ui/icons/CloseIcon'
import InlineLoader from '~/ui/layout/InlineLoader'
import MoveArrowIcon from '~/ui/icons/MoveArrowIcon'
import MoveHelperModal from '~/ui/users/MoveHelperModal'
import Tooltip from '~/ui/global/Tooltip'
import {
  StyledSnackbar,
  StyledSnackbarContent,
  SnackbarBackground,
  StyledSnackbarText,
} from '~/ui/global/styled/material-ui'

const IconHolder = styled.span`
  margin-left: 40px;
  margin-top: 8px;
  width: 16px;

  button {
    height: 19px;
    width: 16px;
  }
`

const CloseIconHolder = styled.span`
  margin-left: 60px;
  width: 16px;

  button {
    height: 19px;
    width: 16px;
  }
`

@inject('uiStore', 'apiStore')
@observer
class MoveModal extends React.Component {
  @observable
  isLoading = false

  handleClose = ev => {
    ev.preventDefault()
    const { uiStore } = this.props
    // Notify the user if they're on a different collection
    if (
      uiStore.viewingCollection &&
      uiStore.movingFromCollectionId !== uiStore.viewingCollection.id
    ) {
      if (uiStore.cardAction === 'move') {
        uiStore.alert(
          'Your items have been returned to their original location',
          'Back'
        )
      }
    }
    uiStore.closeMoveMenu()
  }

  moveCards = async placement => {
    const { uiStore, apiStore } = this.props
    const { viewingCollection, cardAction } = uiStore
    // Viewing collection might not be set, such as on the search page
    if (!viewingCollection) {
      uiStore.alert("You can't move an item here")
      return
    }
    const collectionId = viewingCollection.id
    const movingFromCollection = apiStore.find(
      'collections',
      uiStore.movingFromCollectionId
    )
    const error = this.moveErrors({
      viewingCollection,
      movingFromCollection,
      cardAction,
    })
    if (error) {
      uiStore.alert(error)
      return
    }
    let data = {
      to_id: collectionId,
      from_id: uiStore.movingFromCollectionId,
      collection_card_ids: uiStore.movingCardIds,
      placement,
    }
    try {
      runInAction(() => {
        this.isLoading = true
      })
      let successMessage
      switch (cardAction) {
        case 'move':
          await apiStore.request('collection_cards/move', 'PATCH', data)
          successMessage = 'Items successfully moved!'
          break
        case 'link':
          await apiStore.request('collection_cards/link', 'POST', data)
          successMessage = 'Items successfully linked!'
          break
        case 'duplicate':
          await apiStore.request('collection_cards/duplicate', 'POST', data)
          // have to re-fetch here because the duplicate method wasn't re-rendering
          // see note in collection_cards_controller#duplicate
          await apiStore.request(`collections/${collectionId}`)
          successMessage = 'Items successfully duplicated!'
          break
        case 'useTemplate': {
          data = {
            parent_id: data.to_id,
            template_id: data.from_id,
            placement,
          }
          await apiStore.createTemplateInstance(data)
          successMessage = 'Your template instance has been created!'
          // refresh the current collection to get the new template
          await apiStore.request(`collections/${collectionId}`)
          break
        }
        default:
          return
      }
      runInAction(() => {
        this.isLoading = false
      })
      uiStore.alertOk(successMessage)
      uiStore.resetSelectionAndBCT()
      uiStore.closeMoveMenu()
      if (placement === 'beginning') {
        uiStore.scroll.scrollToTop()
      } else {
        uiStore.scroll.scrollToBottom()
      }
    } catch (e) {
      runInAction(() => {
        this.isLoading = false
      })
      uiStore.alert('You cannot move a collection within itself')
    }
  }

  moveErrors = ({ viewingCollection, movingFromCollection, cardAction }) => {
    if (!viewingCollection.can_edit_content) {
      return "You don't have permission to move items to this collection"
    } else if (
      // don't allow moving cards from templates to non-templates
      cardAction === 'move' &&
      movingFromCollection &&
      movingFromCollection.isMasterTemplate &&
      !viewingCollection.isMasterTemplate
    ) {
      return "You can't move pinned template items out of a template"
    } else if (
      cardAction === 'useTemplate' &&
      viewingCollection.id === movingFromCollection.id
    ) {
      return "You can't create a template inside itself"
    }
    return ''
  }

  handleMoveToBeginning = () => {
    this.moveCards('beginning')
  }

  handleMoveToEnd = () => {
    this.moveCards('end')
  }

  get moveMessage() {
    const { uiStore } = this.props
    const { cardAction, templateName } = uiStore
    const amount = uiStore.movingCardIds.length
    let message = ''
    if (cardAction === 'move') {
      message = `${amount} in transit`
    } else if (cardAction === 'useTemplate') {
      message = `${templateName} in transit`
    } else {
      message = `${amount} selected to ${cardAction}`
    }
    return message
  }

  get moveHelper() {
    const { uiStore, apiStore } = this.props
    const { cardAction, templateName } = uiStore
    const helperProps = {
      type: 'move',
    }
    if (cardAction === 'useTemplate') {
      if (!apiStore.currentUser.show_template_helper) {
        return null
      }
      helperProps.recordName = templateName
      helperProps.type = 'template'
    } else if (!apiStore.currentUser.show_move_helper) {
      return null
    }
    return (
      <MoveHelperModal currentUser={apiStore.currentUser} {...helperProps} />
    )
  }

  render() {
    const { uiStore } = this.props

    return (
      <div>
        {uiStore.movingCardIds.length > 0 && (
          <div>
            <StyledSnackbar classes={{ root: 'Snackbar' }} open>
              {this.isLoading ? (
                <SnackbarBackground>
                  <InlineLoader />
                </SnackbarBackground>
              ) : (
                <StyledSnackbarContent
                  classes={{ root: 'SnackbarContent' }}
                  message={
                    <StyledSnackbarText id="message-id">
                      {this.moveMessage}
                    </StyledSnackbarText>
                  }
                  action={[
                    <IconHolder key="moveup">
                      <Tooltip
                        classes={{ tooltip: 'Tooltip' }}
                        title="Place at top"
                        placement="top"
                      >
                        <button onClick={this.handleMoveToBeginning}>
                          <MoveArrowIcon direction="up" />
                        </button>
                      </Tooltip>
                    </IconHolder>,
                    <IconHolder key="movedown">
                      <Tooltip
                        classes={{ tooltip: 'Tooltip' }}
                        title="Place at bottom"
                        placement="top"
                      >
                        <button onClick={this.handleMoveToEnd}>
                          <MoveArrowIcon direction="down" />
                        </button>
                      </Tooltip>
                    </IconHolder>,
                    <CloseIconHolder key="close">
                      <Tooltip
                        classes={{ tooltip: 'Tooltip' }}
                        title="Cancel"
                        placement="top"
                      >
                        <button onClick={this.handleClose}>
                          <CloseIcon />
                        </button>
                      </Tooltip>
                    </CloseIconHolder>,
                  ]}
                />
              )}
            </StyledSnackbar>
            {this.moveHelper}
          </div>
        )}
      </div>
    )
  }
}

MoveModal.propTypes = {}
MoveModal.wrappedComponent.propTypes = {
  apiStore: MobxPropTypes.objectOrObservableObject.isRequired,
  uiStore: MobxPropTypes.objectOrObservableObject.isRequired,
}
MoveModal.defaultProps = {}

export default MoveModal
