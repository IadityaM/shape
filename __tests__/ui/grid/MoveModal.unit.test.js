import MoveModal from '~/ui/grid/MoveModal'
import MoveHelperModal from '~/ui/users/MoveHelperModal'
import { fakeCollection, fakeItemCard } from '#/mocks/data'
import expectTreeToMatchSnapshot from '#/helpers/expectTreeToMatchSnapshot'

import fakeApiStore from '#/mocks/fakeApiStore'
import fakeUiStore from '#/mocks/fakeUiStore'

let props, wrapper, component
const uiStore = fakeUiStore
describe('MoveModal', () => {
  beforeEach(() => {
    uiStore.viewingCollection = {
      id: 3,
      API_fetchCards: jest.fn(),
      movingCardIds: [10],
    }
    props = {
      apiStore: fakeApiStore({
        requestResult: { data: fakeCollection },
        findResult: fakeItemCard,
      }),
      uiStore,
    }
    props.apiStore.request = jest.fn()
    props.uiStore.alert.mockClear()
    props.uiStore.scrollToTop.mockClear()
    wrapper = shallow(<MoveModal.wrappedComponent {...props} />)
    component = wrapper.instance()
  })

  it('renders snapshot', () => {
    expectTreeToMatchSnapshot(wrapper)
  })

  describe('handleClose', () => {
    const fakeEvent = {
      preventDefault: jest.fn(),
    }

    it('should close the move menu in the uiStore', () => {
      component.handleClose(fakeEvent)
      expect(props.uiStore.closeMoveMenu).toHaveBeenCalled()
    })
  })

  describe('moveHelper', () => {
    describe('template helper', () => {
      beforeEach(() => {
        props.apiStore.currentUser.show_template_helper = true
        props.uiStore.cardAction = 'useTemplate'
        wrapper = shallow(<MoveModal.wrappedComponent {...props} />)
      })
      it('should display if the user should see the template helper', () => {
        expect(wrapper.find(MoveHelperModal).exists()).toBeTruthy()
      })
    })
    describe('move helper', () => {
      beforeEach(() => {
        props.apiStore.currentUser.show_move_helper = true
        props.uiStore.cardAction = 'move'
        wrapper = shallow(<MoveModal.wrappedComponent {...props} />)
      })
      it('should display if the user should see the move helper', () => {
        expect(wrapper.find(MoveHelperModal).exists()).toBeTruthy()
      })
    })

    describe('dismissed helper', () => {
      beforeEach(() => {
        props.uiStore.dismissedMoveHelper = true
      })
      it('should not display the helper', () => {
        expect(wrapper.find(MoveHelperModal).exists()).toBeFalsy()
      })
    })
  })
})
