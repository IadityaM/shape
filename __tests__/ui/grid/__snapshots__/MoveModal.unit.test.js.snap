// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoveModal renders snapshot 1`] = `
<div
  style={
    Object {
      "display": "none",
    }
  }
>
  <StyledSnackbar
    classes={
      Object {
        "root": "Snackbar",
      }
    }
    open={true}
  >
    <React.Fragment>
      <StyledSnackbarContent
        action={
          Array [
            <styled.span>
              <Tooltip
                classes={
                  Object {
                    "tooltip": "Tooltip",
                  }
                }
                placement="top"
                title="Place at top"
              >
                <button
                  data-cy="MoveModalArrow-up"
                  onClick={[Function]}
                >
                  <ArrowIcon
                    direction="up"
                  />
                </button>
              </Tooltip>
            </styled.span>,
            <styled.span>
              <Tooltip
                classes={
                  Object {
                    "tooltip": "Tooltip",
                  }
                }
                placement="top"
                title="Place at bottom"
              >
                <button
                  data-cy="MoveModalArrow-down"
                  onClick={[Function]}
                >
                  <ArrowIcon
                    direction="down"
                  />
                </button>
              </Tooltip>
            </styled.span>,
            <styled.span>
              <Tooltip
                classes={
                  Object {
                    "tooltip": "Tooltip",
                  }
                }
                placement="top"
                title="Cancel"
              >
                <button
                  onClick={[Function]}
                >
                  <CloseIcon />
                </button>
              </Tooltip>
            </styled.span>,
          ]
        }
        classes={
          Object {
            "root": "SnackbarContent",
          }
        }
        message={
          <StyledSnackbarText
            id="message-id"
          >
            0 card in transit
          </StyledSnackbarText>
        }
      />
    </React.Fragment>
  </StyledSnackbar>
</div>
`;
